### PR TITLE
fix: Falling Off Edge in Pet Puzzle

### PR DESCRIPTION
**Pull request by TAHuntling**
Issue #547 
Description: Modified PetComponents.cpp, dNavMesh.cpp and dNavMesh.h to include a new function nearest point which helps to prevent players from falling off a cliff or an edge when taming a pet. It will instead try to keep the player on the nearest navMesh (terrain)

Motivation and Context: Linked issue from EmosewaMC to work on

Type of Changes: Added new function to dNavMesh and its respective includes in dNavMesh.h. Made changes to petcomponent OnUse (which is used to start a player taming pet interaction).

How Has This Been Tested? : Taming a pet, such as the Goat in Forbidden Valley while on the edge of a cliff, which resulted in keeping the player as close to the edge as necessary to prevent them from falling off. This allows (assuming the camera angle and perspective does not get wacky, which is another WIP). Also testing taming pets right next to ledges, which prevents falling through the wall or clipping through the wall as previously reported.

### CODE REVIEW

File: dGame/dComponents/PetComponent.cpp [hunk 1]
```diff
@@ -246,23 +247,30 @@ void PetComponent::OnUse(Entity* originator) {
...
 		originator->GetObjectID(),
-		true,
+		false,
 		ePetTamingNotifyType::BEGIN,
-		petPosition,
-		position,
-		rotation,
+		NiPoint3Constant::ZERO,
+		NiPoint3Constant::ZERO,
+		NiQuaternion(0.0f, 0.0f, 0.0f, 0.0f),
```
Comment (EmosewaMC) [inline dGame/dComponents/PetComponent.cpp:254]: Yes, once again lu uses NiQuaternion 0 and not identity.  Fantastic game

File: dNavigation/dNavMesh.cpp [hunk 2]
```diff
@@ -112,6 +112,34 @@ void dNavMesh::LoadNavmesh() {
...
+
+		dtPolyRef nearestRef = 0;
+		float polyPickExt[3] = { halfExtent, halfExtent, halfExtent };
+		float nearestPoint[3] = { 0.0f, 0.0f, 0.0f };
+		dtQueryFilter filter{};
+
+		auto hasPoly = m_NavQuery->findNearestPoly(pos, polyPickExt, &filter, &nearestRef, nearestPoint);
+		toReturn.x = nearestPoint[0];
+		toReturn.y = nearestPoint[1];
+		toReturn.z = nearestPoint[2];
```
Comment (EmosewaMC) [inline dNavigation/dNavMesh.cpp:131]:
```suggestion
```

Comment (EmosewaMC) [reply dNavigation/dNavMesh.cpp:131]: seems to be extra code and isnt necessary given the if chain below

Comment (TAHuntling) [reply dNavigation/dNavMesh.cpp]: Removed redundant code

**Review by aronwk-aaron — APPROVED** [review]

**Review by EmosewaMC — APPROVED** [review]

**Pull request merged**